Client/Frontend/Browser/BrowserViewController/BrowserViewController+URLBarDelegate.swift

```diff
@@ -12,7 +12,14 @@ protocol OnViewDismissable: class {
 
 class DismissableNavigationViewController: UINavigationController, OnViewDismissable {
     var onViewDismissed: (() -> Void)? = nil
-
+    var onViewWillDisappear: (() -> Void)? = nil
```

Comment:
Do we need both `onViewDismissed` and `onViewWillDisappear`? 🤔

Reply:
When I tried onViewDIsmissed alone, the user would see the section disappear (if they delete all bookmarks from the bookmarks panel). onViewWillDisappear helps to avoid that.

But having onViewWillDisappear alone doesn't work. If a user begins to slide down on the modal but then decides not to, there can be a case where the section should have disappeared (b/c all bookmarks are deleted). In that case, the cells and section remain. When scrolled offscreen, the section remains with a big empty space. 

Having both helped to avoid these odd problems. But I'm open to suggestions for better ways!